src/basic/fileio.c

```diff
@@ -398,21 +404,21 @@ int read_full_virtual_file(const char *filename, char **ret_contents, size_t *re
                 /* Be prepared for files from /proc which generally report a file size of 0. */
                 assert_cc(READ_FULL_BYTES_MAX < SSIZE_MAX);
                 if (st.st_size > 0) {
-                        if (st.st_size > READ_FULL_BYTES_MAX)
+                        size = MIN((size_t) st.st_size, max_size);
```

Comment:
the could possibly overflow on 32bit userspace on a 64bit kernel (i.e you run this on /proc/kcore), because the file size is going to be 64bit, but size_t is 32bit.